module-system/module-implementations/sov-evm/src/tests/genesis_tests.rs

```diff
@@ -0,0 +1,127 @@
+use reth_rpc_types::{Block, BlockTransactions, Header};
+use revm::primitives::{SpecId, KECCAK_EMPTY, U256};
+use sov_modules_api::default_context::DefaultContext;
+use sov_modules_api::Module;
+use sov_state::{DefaultStorageSpec, ProverStorage, WorkingSet};
+
+// use crate::evm::db;
+use crate::{evm::EvmChainConfig, AccountData, Evm, EvmConfig, SpecIdWrapper};
+type C = DefaultContext;
+
+#[test]
+fn genesis_data() {
+    get_evm(None);
+
+    // TODO: assert on account being the same - easier after unifying stored types!
+    // let db_account = evm.accounts.get(&address, working_set).unwrap();
+}
+
+#[test]
+fn genesis_cfg() {
+    let (evm, mut working_set) = get_evm(None);
+
+    let cfg = evm.cfg.get(&mut working_set).unwrap();
+    assert_eq!(
+        cfg,
+        EvmChainConfig {
+            spec: vec![
+                (0, SpecIdWrapper(SpecId::BERLIN)),
+                (1, SpecIdWrapper(SpecId::LATEST))
+            ],
+            chain_id: 1000,
+            block_gas_limit: reth_primitives::constants::ETHEREUM_BLOCK_GAS_LIMIT,
+            block_timestamp_delta: 2,
+            coinbase: [3u8; 20],
+            limit_contract_code_size: Some(5000)
+        }
+    );
+}
+
+#[test]
+#[should_panic(expected = "EVM spec must start from block 0")]
+fn genesis_cfg_missing_specs() {
+    get_evm(Some(EvmConfig {
+        spec: vec![(5, SpecId::BERLIN)].into_iter().collect(),
+        ..Default::default()
+    }));
+}
+
+#[test]
+fn genesis_block() {
+    let (evm, mut working_set) = get_evm(None);
+
+    let block = evm
+        .blocks
+        .get(&0u64, &mut working_set.accessory_state())
+        .unwrap();
+
+    assert_eq!(
+        block,
+        Block {
+            header: Header {
+                parent_hash: H256::default(),
+                state_root: KECCAK_EMPTY,
+                transactions_root: EMPTY_TRANSACTIONS,
+                receipts_root: EMPTY_RECEIPTS,
+                logs_bloom: Bloom::default(),
+                difficulty: U256::ZERO,
+                number: Some(U256::ZERO),
+                gas_limit: U256::from(ETHEREUM_BLOCK_GAS_LIMIT),
+                gas_used: U256::ZERO,
+                timestamp: U256::from(50),
+                extra_data: Bytes::default(),
+                mix_hash: H256::default(),
+                nonce: Some(H64::default()),
+                base_fee_per_gas: Some(U256::from(70)),
+                hash: Some(H256(hex!(
+                    "d57423e4375c45bc114cd137146aab671dbd3f6304f05b31bdd416301b4a99f0"
+                ))),
+                uncles_hash: EMPTY_OMMER_ROOT,
+                miner: Address::from([3u8; 20]),
+                withdrawals_root: None
+            },
+            transactions: BlockTransactions::Hashes(vec![]),
+            total_difficulty: None,
+            uncles: vec![],
+            size: None,
+            withdrawals: None
+        }
+    );
+}
+
+pub(crate) fn get_evm(
+    config: Option<EvmConfig>,
```

Comment:
Nit (not important): we could also make a global test variable `test_config` and remove the `Option` here, then the caller would explicitly pass "custom_config"  or `global_test_config`.  This way we move the hardcoded data outside of the function.

Reply:
used `lazy_static` as discussed